### PR TITLE
Added unit tests for status and cronfederatedhpa helper utilities

### PR DESCRIPTION
**Pull request by anujagrawal699**
**Description:**
This PR introduces comprehensive unit tests for status and cronfederatedhpa helper utility functions in the pkg/util/helper. These additions aim to increase code reliability, maintainability, and test coverage of the helper utility package.

**Additions:**
1. pkg/util/helper/cronfederatedhpa_test.go
2. pkg/util/helper/status_test.go

**Test Coverage:**
1. pkg/util/helper/cronfederatedhpa_test.go : 0% to 100%
2. pkg/util/helper/status_test.go : 0% to 100%

**What type of PR is this?**
/kind feature

**Which issue(s) this PR fixes**:
Fixes a part of #5470 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/util/helper/status_test.go [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+func TestMutate(t *testing.T) {
+	tests := []struct {
+		name          string
+		key           types.NamespacedName
+		obj           *corev1.Pod
+		mutateFn      controllerutil.MutateFn
+		expectedError string
+	}{
+		{
```
Comment (zhzhuang-zju) [inline pkg/util/helper/status_test.go:193]: Can you add test cases for `mutating object name/namespace`? Although `func TestUpdateStatus` does cover related scenarios, as a unit test, individual test functions should aim to measure as comprehensive a range of scenarios as possible

**Review by zhzhuang-zju — COMMENTED** [review]
others LGTM

File: pkg/util/helper/status_test.go (hunk 1 above)
Comment (anujagrawal699) [reply pkg/util/helper/status_test.go:193]: Yes, i'll add it.

File: pkg/util/helper/status_test.go [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+			newKey := types.NamespacedName{
+				Name:      testObj.GetName(),
+				Namespace: testObj.GetNamespace(),
+			}
+
+			if tt.expectedError == "" {
+				assert.Equal(t, tt.key, newKey)
+				// For successful mutations, verify other metadata changes were applied
+				if testObj.GetLabels() != nil {
+					assert.NotEmpty(t, testObj.GetLabels())
```
[the remark this answers is not in the document]
Comment (anujagrawal699) [reply pkg/util/helper/status_test.go:339]: Yes

File: pkg/util/helper/status_test.go [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+			}
+
+			// Verify the object key hasn't changed for successful mutations
+			newKey := types.NamespacedName{
+				Name:      testObj.GetName(),
+				Namespace: testObj.GetNamespace(),
+			}
+
+			if tt.expectedError == "" {
+				assert.Equal(t, tt.key, newKey)
```
[the remark this answers is not in the document]
Comment (anujagrawal699) [reply pkg/util/helper/status_test.go:336]: For error cases, we still want to verify that the object wasn't modified incorrectly. For example, when we try to modify the name/namespace and get an error, we want to verify that the name/namespace didn't actually change. We're already using `tt.expectedError == ""` to guard the assertions that should only run for success cases.

[the remark this answers is not in the document]
Comment (anujagrawal699) [reply pkg/util/helper/status_test.go:336]: I'll remove the `assert.Equal(t, tt.key, newKey) from the if case.

File: pkg/util/helper/status_test.go [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+			}
+			assert.NoError(t, err)
+
+			// Verify the object key hasn't changed for successful mutations
+			newKey := types.NamespacedName{
+				Name:      testObj.GetName(),
+				Namespace: testObj.GetNamespace(),
+			}
+
+			if tt.expectedError == "" {
```
Comment (zhzhuang-zju) [inline pkg/util/helper/status_test.go:335]:
```suggestion
```

Lines `L322-L327` have already handled the case where `expectedError` is not empty, so there is no need to add a condition check for `tt.expectedError == ""`.

**Review by zhzhuang-zju — COMMENTED** [review]
thanks, others LGTM

**Comment by zhzhuang-zju** [comment]
/lgtm
cc @XiShanYongYe-Chang

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/approve

File: pkg/util/helper/cronfederatedhpa_test.go [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+		rule     autoscalingv1alpha1.CronFederatedHPARule
+		expected int
+	}{
+		{
+			name:     "successful history limit is nil",
+			rule:     autoscalingv1alpha1.CronFederatedHPARule{},
+			expected: 3,
+		},
+		{
+			name: "successful history limit is set to 5",
```
[the remark this answers is not in the document]
Comment (anujagrawal699) [reply pkg/util/helper/cronfederatedhpa_test.go:76]: Yeah that makes sense, 3 could be defined as a constant at the package level, which would make it easier to change the default value in the future if needed.

**Pull request merged**